program/mobilenets-armcl-opencl/armcl_graph_common.h

```diff
@@ -0,0 +1,122 @@
+  arm_compute::CLScheduler::get().default_init(cl_tuner);
+
+  // Should be called after initialization
+  set_kernel_path();
+
+#if defined(ARMCL_18_05_PLUS)
+  arm_compute::graph::backends::BackendRegistry::get()
+    .add_backend<arm_compute::graph::backends::CLDeviceBackend>(
+      arm_compute::graph::Target::CL);
+#endif
+}
+
+#if defined(ARMCL_18_05_PLUS)
+
+inline arm_compute::graph::DepthwiseConvolutionMethod str_to_dwsc_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "DEFAULT") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::DEFAULT;
+  if (strcmp(method_name, "GEMV") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::GEMV;
+  if (strcmp(method_name, "OPTIMIZED_3x3") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::OPTIMIZED_3x3;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::DepthwiseConvolutionMethod>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethod str_to_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "DEFAULT") == 0) return arm_compute::graph::ConvolutionMethod::DEFAULT;
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethod::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethod::DIRECT;
+  if (strcmp(method_name, "WINOGRAD") == 0) return arm_compute::graph::ConvolutionMethod::WINOGRAD;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::ConvolutionMethod>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethod get_convolution_method() {
+  auto method_name = getenv("CK_CONVOLUTION_METHOD");
+  if (!method_name) {
+      bool bifrost_target = (arm_compute::CLScheduler::get().target() == arm_compute::GPUTarget::BIFROST);
+      return (bifrost_target ? arm_compute::graph::ConvolutionMethod::DIRECT : arm_compute::graph::ConvolutionMethod::GEMM);
+  }
+  return str_to_convolution_method(method_name);
+}
+
+inline arm_compute::graph::Target get_target_hint() {
+  return arm_compute::graph::Target::CL;
+}
+
+#define GRAPH(graph_var, graph_name)\
+  arm_compute::graph::frontend::Stream graph_var{ 0, graph_name };
+
+#else // ArmCL < 18.05
+
+inline arm_compute::graph::ConvolutionMethodHint str_to_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethodHint::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethodHint::DIRECT;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::ConvolutionMethodHint>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethodHint get_convolution_method() {
```

Comment:
Isn't this the same as for the case `ArmCL >= 18.05`?

Reply:
When `ArmCL >= 18.05` then `arm_compute::graph::ConvolutionMethod` is used, but not `arm_compute::graph::ConvolutionMethodHint`